src/predict.jl

```diff
@@ -0,0 +1,179 @@
+                                                    levels.(mnew.reterms))
+            if sort!(known_levels) != sort!(data_levels)
+                throw(ArgumentError("New level enountered in $grp"))
+            end
+        end
+
+        # we don't have to worry about the BLUP ordering within a given
+        # grouping variable because we are in the :error branch
+        blups = ranef(m)[oldreperm]
+    elseif new_re_levels == :population
+        blups = ranef(mnew)[newreperm]
+        blupsold = ranef(m)[oldreperm]
+
+        for (idx, B) in enumerate(blups)
+            oldlevels = levels(oldre[idx])
+            for (lidx, ll) in enumerate(levels(newre[idx]))
+                oldloc = findfirst(isequal(ll), oldlevels)
+                if oldloc === nothing
+                    # setting a BLUP to zero gives you the population value
+                    B[:, lidx] .= zero(T)
+                else
+                    B[:, lidx] .= @view blupsold[idx][:, oldloc]
+                end
+            end
+        end
+    elseif new_re_levels == :missing
+        # we can't quite use ranef! because we need
+        # Union{T, Missing} and not just T
+        blups = Vector{Matrix{Union{T,Missing}}}(undef, length(m.reterms))
+        copyto!(blups, ranef(mnew)[newreperm])
+        blupsold = ranef(m)[oldreperm]
+        for (idx, B) in enumerate(blups)
+            oldlevels = levels(oldre[idx])
+            for (lidx, ll) in enumerate(levels(newre[idx]))
+                oldloc = findfirst(isequal(ll), oldlevels)
+                if oldloc === nothing
+                    # missing is poisonous so propogates
+                    B[:, lidx] .= missing
+                else
+                    B[:, lidx] .= @view blupsold[idx][:, oldloc]
+                end
+            end
+        end
+    # elseif new_re_levels == :simulate
+    #     @show m.θ
+    #     updateL!(setθ!(mnew, m.θ))
+    #     blups = ranef(mnew)[newreperm]
+    #     blupsold = ranef(m)[oldreperm]
+    #     for (idx, B) in enumerate(blups)
+    #         oldlevels = levels(oldre[idx])
+    #         for (lidx, ll) in enumerate(levels(newre[idx]))
+    #             oldloc = findfirst(isequal(ll), oldlevels)
+    #             if oldloc === nothing
+    #                 # keep the new value
+    #             else
+    #                 B[:, lidx] = @view blupsold[idx][:, oldloc]
+    #             end
+    #         end
+    #     end
+    else
+        throw(ErrorException("Impossible branch reached. Please report an issue on GitHub"))
+    end
+
+    for (rt, bb) in zip(newre, blups)
+        mul!(y, rt, bb, one(T), one(T))
+    end
+
+    y
+end
+
+function StatsBase.predict(m::GeneralizedLinearMixedModel{T}, newdata::Tables.ColumnTable;
+                           new_re_levels=:population, type=:response) where T
+    type in (:linpred, :response) || throw(ArgumentError("Invalid value for type: $(type)"))
+
+    y = predict(m.LMM, newdata; new_re_levels)
```

Comment:
@dmbates  I need to refactor this a bit so that I can share the BLUP extraction logic between both LMM and GLMM, but this isn't the correct way to do it.

Reply:
I think I've got it figured out, testing locally now

Reply:
Okay.  I'm just starting to read through all this again.  If you have suggestions of where I should start please let me know.

Reply:
I think I've got it refactored appropriately now and am tracking down some floating-point tolerance issues for `fast=false` fits.

I've pushed the changes so far.